bridges/TwitterBridge.php

```diff
@@ -613,19 +732,144 @@ private function getListId($username, $listName) {
 		return $searchResult->data->user_by_screen_name->list->id_str;
 	}
 
-	private function getUserInformation($userId, $apiData) {
-		foreach($apiData->users as $user) {
-			if($user->id_str == $userId) {
-				return $user;
+	// private function getUserInformation($userId, $apiData) {
+	// 	foreach($apiData->users as $user) {
+	// 		if($user->id_str == $userId) {
+	// 			return $user;
+	// 		}
+	// 	}
+	// }
+
+	// private function getTweet($tweetId, $apiData) {
+	// 	if (property_exists($apiData->tweets, $tweetId)) {
+	// 		return $apiData->tweets->$tweetId;
+	// 	} else {
+	// 		return null;
+	// 	}
+	// }
+
+	/**
+	 * Tries to make an API call to twitter.
+	 * @param $api string API entry point
+	 * @param $params array additional URI parmaeters
+	 * @return object json data
+	 */
+	private function makeApiCall($api, $params) {
+		$uri = self::API_URI . $api . '?' . http_build_query($params);
+
+		$retries = 1;
+		$retry = 0;
+		do {
+			$retry = 0;
+
+			$result = $this->getContents($uri, $this->authHeaders, array(), true);
+
+			switch($result['errorcode']) {
+			case 200: // Contents OK
+			case 201: // Contents Created
+			case 202: // Contents Accepted
+				break;
+
+			case 401:
+			case 403:
+				if ($retries) {
+					$retries--;
+					$retry = 1;
+					$this->getApiKey(1);
+					continue 2;
+				}
+			default:
+				$code = $result['errorcode'];
+				$data = $result['content'];
+				returnServerError(<<<EOD
+Failed to make api call: $api
+HTTP Status: $code
+Errormessage: $data
+EOD
+				);
+				break;
 			}
-		}
+		} while ($retry);
+
+		$data = json_decode($result['content']);
+
+		return $data;
 	}
 
-	private function getTweet($tweetId, $apiData) {
-		if (property_exists($apiData->tweets, $tweetId)) {
-			return $apiData->tweets->$tweetId;
-		} else {
-			return null;
+	private function getContents($url, $header = array(), $opts = array(), $returnHeader = false){
+		Debug::log('Reading contents from "' . $url . '"');
+
```

Comment:
What is the difference with original getContents?

Reply:
Main reason is, that it don't throw a exception when http status code is >= 400.  Twitter adds a json coded error message inside the http response body. When using the version from the lib, this information is lost. 
This was a quick hack to prevent changes outside of twitterbridge. On long term, it would be better to adjust the lib version. 

I can adjust the lib version and add there a plain version, without status code evaluation and throwing exception. Extract the curl code into a separate function, which can be called directly and will be called from the original getContents function.
Or would it be better to make this in a separate PR ? I'm unfamiliar, what is preferred by other developers of the project.

Reply:
Made separate PR. https://github.com/RSS-Bridge/rss-bridge/pull/2447

Reply:
Suggesting following patch to adapt changes from 2447
https://pastebin.com/Z7ywpPDE

Reply:
Please ignore must patch. It is not valid. In any case, could you please adapt your bridge to latest changes in getContents that raises UnexpectedResponse exception?